Sources/ArrayResponse.swift

```diff
@@ -52,4 +67,31 @@ public struct ArrayResponse<T: Resource>: ImmutableMappable {
             entry.resolveLinks(against: allIncludedEntries, and: (includedAssets ?? []))
         }
     }
+
+}
+
+// TODO: Document.
+public struct MappedArrayResponse<ItemType>: Array where ItemType: EntryModellable {
```

Comment:
The other concrete implementation of `Array` is `ArrayResponse<ItemType>: where ItemType: Resource`.

Resource is the base class for `ContentType` and `Asset`